projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -670,15 +646,17 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     public onFilteringDone = new EventEmitter<IFilteringExpressionsTree>();
 
     /**
-     * Emitted after paging is performed.
+     * @deprecated Use `IgxPaginator` outputs instead.
      *
+     * Emitted after paging is performed.
      * @remarks
      * Returns an object consisting of the previous and next pages.
      * @example
      * ```html
      * <igx-grid #grid [data]="localData" [height]="'305px'" [autoGenerate]="true" (onPagingDone)="pagingDone($event)"></igx-grid>
      * ```
      */
+    @DeprecateProperty('Use the corresponding event emiited by the `igx-paginator`.')
     @Output()
     public onPagingDone = new EventEmitter<IPageEventArgs>();
```

Comment:
We're having the rest of the on-events removed, instead of deprecating them first. You can go ahead and remove the paging events from the grid too.